### DISCUSSION TITLE
how does the BroadPhase quadtree works?

### DISCUSSION THREAD

**Discussion by SirLynix** [jrouwe/JoltPhysics#834] [Q&A] [answered]
Hello there,

I was wondering something, since Jolt Physics is a 3D physics engine, why does it uses a quadtree instead of an octree for its BroadPhase? Maybe I'm missing something?

Is it because it's optimized for games with low verticality discrepancy such as Horizon?

**Comment by jrouwe** [comment, marked as answer]
It uses a quadtree because the Intel SSE/ARM NEON instructions allow you to do 4 computations at the same time making it possible to e.g. check a ray vs 4 boxes in 1 calculation.

B.t.w. a quadtree doesn't just split horizontally, it also splits vertically: it calculates the bounding box of the center of all object bounding boxes and splits along the longest axis. 

So there's no real benefit in using an octree.

**Reply by SirLynix** [reply]
Thanks for the clarification. But does that mean it's not suited to a 3D game set in space where all three axis are fully used (for example, bodies can be in [-4km;4km] on XYZ)?

**Reply by jrouwe** [reply]
It is perfectly suitable for a space game, as said the splitting doesn't restrict itself to the XY plane, it splits based on the actual object distribution and every level of the tree can split in a different axis.

**Reply by SirLynix** [reply]
It makes sense, thank you!
